controllers/autohealconfig_controller_test.go

```diff
@@ -0,0 +1,693 @@
+				Client: tt.fields.Client,
+			}
+			repoSecret, err := r.getConfigRepoSecret(tt.args.ctx, tt.args.autoheal, tt.args.status)
+			if (err != nil) != tt.err {
+				t.Errorf("GetConfigRepoSecret() error = %v wantedError= %v", err, tt.err)
+
+			}
+			if !reflect.DeepEqual(repoSecret, tt.want) {
+				t.Errorf("GetConfigRepoSecret() got = %v, want %v", repoSecret, tt.want)
+			}
+		})
+	}
+
+}
+
+func TestReadFiles(t *testing.T) {
+
+	type fields struct {
+		httpClient *httptest.Server
+	}
+
+	type args struct {
+		repo *v1alpha1.ConfigRepositoryInfo
+	}
+	mockGood := mockHttpClientGood(t)
+	mockBad := mockHttpClientBad(t)
+	mockGoodBad := mockHttpClientGoodBad(t)
+
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   *v1alpha1.AutohealConfigTree
+		err    bool
+	}{
+		{
+			name: "Test repo data is present, correct and struct gets populated",
+			fields: fields{
+				mockGood,
+			},
+			args: args{
+				repo: &v1alpha1.ConfigRepositoryInfo{Tag: tag, RepositoryURL: mockGood.URL, AccessToken: accessToken},
+			},
+			want: &configRepo,
+
+			err: false,
+		},
+		{
+			name: "Bad gateway, struct does not get populated returns nothing, error thrown",
+			fields: fields{
+				mockBad,
+			},
+			args: args{
+				repo: &v1alpha1.ConfigRepositoryInfo{Tag: tag, RepositoryURL: mockBad.URL, AccessToken: accessToken},
+			},
+			want: nil,
+			err:  false,
+		},
+		{
+			name: "Cant make new request returns nothing, error thrown",
+			fields: fields{
+				mockBad,
+			},
+			args: args{
+				repo: &v1alpha1.ConfigRepositoryInfo{Tag: tag, RepositoryURL: repoUrl, AccessToken: accessToken},
+			},
+			want: nil,
+			err:  true,
+		},
+		{
+			name: "Cant read request body returns nothing, error thrown",
+			fields: fields{
+				mockGoodBad,
+			},
+			args: args{
+				repo: &v1alpha1.ConfigRepositoryInfo{Tag: tag, RepositoryURL: mockGoodBad.URL, AccessToken: accessToken},
+			},
+			want: nil,
+			err:  true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r := AutohealConfigReconciler{
+				httpClient: *tt.fields.httpClient.Client(),
+			}
+			files, err := r.readFiles(tt.args.repo)
+			if (err != nil) != tt.err {
+
+				t.Errorf("readFiles() error = %v wantedError= %v", err, tt.err)
+
+			}
+			if !reflect.DeepEqual(files, tt.want) {
+
+				t.Errorf("readFiles() got = %v want %v", files, tt.want)
+			}
+		})
+	}
+}
+
+func TestGetScenarios(t *testing.T) {
+	mockGoodFiles := mockHttpClientGoodFiles(t)
+	mockBad := mockHttpClientBad(t)
+	mockGoodBad := mockHttpClientGoodBad(t)
+
+	type fields struct {
+		httpClient *httptest.Server
+	}
+
+	type args struct {
+		repo     *v1alpha1.ConfigRepositoryInfo
+		files    *v1alpha1.AutohealConfigTree
+		autoheal *v1alpha1.AutohealConfig
+	}
+
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   []v1alpha1.AutohealConfigScenario
+		err    bool
+	}{
+		{
+			name: "Get files and populate struct reruens scenerio struct",
```

Comment:
```suggestion
			name: "Get files and populate struct. Returns a scenario struct",
```
I think that is correct?